plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal.hs

```diff
@@ -277,6 +300,35 @@ fieldSuggestCodeAction recorder ide _ (CodeActionParams _ _ (TextDocumentIdentif
       let completionTexts = fmap (^. JL.label) completions
       pure $ FieldSuggest.fieldErrorAction uri fieldName completionTexts _range
 
+cabalAddCodeAction :: Recorder (WithPriority Log) -> PluginMethodHandler IdeState 'LSP.Method_TextDocumentCodeAction
+cabalAddCodeAction recorder state plId (CodeActionParams _ _ (TextDocumentIdentifier uri) _ CodeActionContext{_diagnostics=diags}) = do
+  maxCompls <- fmap maxCompletions . liftIO $ runAction "cabal.cabal-add" state getClientConfigAction
+  let suggestions = concatMap (\diag -> CabalAdd.hiddenPackageSuggestion maxCompls diag) diags
```

Comment:
```suggestion
  let suggestions = take maxCompls $ concatMap (\diag -> CabalAdd.hiddenPackageSuggestion maxCompls diag) diags
```

Otherwise, you may produce `maxCompls * length diags` number of code actions. Or is that fine?
If you call `take maxCompls` here, you can drop it from `hiddenPackageSuggestion`.

Reply:
Yes, it was not the desired behavior.